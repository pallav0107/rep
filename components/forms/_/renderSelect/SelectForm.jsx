import React from 'react';
import PropTypes from 'prop-types';
import { AlertText } from '@wdpui/react-gel';
import { ErrorIcon, Styledselect } from '../../../../style/style';
import { Box } from 'grid-styled';

const SelectForm = ({
  input,
  meta: { touched, error },
  options,
  id,
  disabled,
}) => (
  <Box>
    <Styledselect
      {...input}
      alertType={touched && error ? 'danger' : null}
      disabled={disabled}
    >
      <option value="">Please Select</option>
      {id !== 'CountryFlag' &&
        options.map(key => (
          <option key={key} value={key}>
            {key}
          </option>
        ))}
      {id === 'CountryFlag' &&
        options.map(key => (
          <option className="in flag" key={key} value={key} />
        ))}
    </Styledselect>
    <span>
      {touched && error && (
        <AlertText styling="danger">
          {' '}
          <ErrorIcon size="xsmall" /> Please select a value! It is a {error}{' '}
          Field.
        </AlertText>
      )}
    </span>
  </Box>
);

SelectForm.propTypes = {
  options: PropTypes.shape([]).isRequired,
  id: PropTypes.string,
  disabled: PropTypes.bool,
};

SelectForm.defaultProps = {
  id: null,
  disabled: false,
};

export default SelectForm;
