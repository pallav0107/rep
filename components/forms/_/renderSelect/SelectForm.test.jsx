import React from 'react';
import { shallow } from 'enzyme';
import SelectForm from './SelectForm';
import { withTheme } from '../../../../../utils/jest/TestUtils';
import "../../../../setupTests"

describe('SelectForm', () => {
  it('should render', () => {
    const component = withTheme(
      <SelectForm options={['Mr', 'Mrs', 'Miss']} meta={['touched', 'error']} />
    );
    const wrapper = shallow(component).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
