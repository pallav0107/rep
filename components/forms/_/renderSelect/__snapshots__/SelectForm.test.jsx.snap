// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectForm should render 1`] = `
<Box>
  <Styled(Select)
    alertType={null}
    disabled={false}
  >
    <option
      value=""
    >
      Please Select
    </option>
    <option
      key="Mr"
      value="Mr"
    >
      Mr
    </option>
    <option
      key="Mrs"
      value="Mrs"
    >
      Mrs
    </option>
    <option
      key="Miss"
      value="Miss"
    >
      Miss
    </option>
  </Styled(Select)>
  <span />
</Box>
`;
